File: verve patch/base/Applications/CSymphony_0_1/vbSamples/CodeIterationStatement.vb
Option Strict Off
Option Explicit On

Imports System
Imports System.CodeDom
Imports System.CodeDom.Compiler
Imports System.Collections
Imports CSymphony.Lib

Namespace CSymphony.CSy
	Public Class CSyUnit
		 Inherits GraphUnit
		Overrides Public Sub Build()
			ns("MyNameSpace")

			declClass(TypePublic, "MyClass")
			declMethod(MemPublic, "System.Void", "MonMethod")

			Dim testInt As VarRef = declVar("System.Int32", "testInt", newLit(0))
			
			forOn(equate (testInt.Expr(), newLit(1)))
			forTest(oper(testInt.Expr(), lt, newLit(10)))
			forInc(oper(testInt.Expr(), add, newLit(1)))
			hi ("for statements come here")
			
			Dim writeLine As MethodRef  = New MethodRef ("System.Console", "WriteLine")
			Dim testIntToStr As MethodRef = New MethodRef (testInt.Expr(), "ToString")

			enter (ex (writeLine, ex(testIntToStr)))
			forOff()
	
		End Sub
	End Class
End Namespace
			

File: verve patch/base/Applications/CSymphony_0_1/csSamples/CodeIterationStatement.cs
using System;
using System.CodeDom;
using System.CodeDom.Compiler;
using System.Collections;
using CSymphony.Lib;

namespace CSymphony.CSy
{
	public class CSyUnit : GraphUnit
	{
		public override void Build()
		{
			ns("MyNameSpace");

			declClass(TypePublic, "MyClass");
			declMethod(MemPublic, "System.Void", "MonMethod");

			VarRef testInt = declVar("System.Int32", "testInt", new Literal (0));
			forOn(equate (testInt, new Literal(1)));
			forTest(testInt < new Literal (10));
			forInc(testInt + new Literal(1));
			hi ("for statements come here");
			MethodRef writeLine = new MethodRef ("System.Console", "WriteLine");
			MethodRef testIntToStr = new MethodRef (testInt, "ToString");

			enter (ex (writeLine, ex(testIntToStr)));
			forOff();
	
		}
	}
}
